### PR TITLE
New review approvals shouldn't require a message

### PR DESCRIPTION
**Pull request by jolheiser**
When reviewing a PR with no previous review comments, approvals should be allowed without a message.

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

**Comment by lunny** [comment]
If you do that, you should also change the comment displayed on issue list and dashboard.

**Comment by jolheiser** [comment]
> If you do that, you should also change the comment displayed on issue list and dashboard.

What do you mean? With this change, nothing shows up on dashboard (do you want it to show there was an approval?)

And the PR looks like this 
gitea-review-approve-1

**Review by zeripath — APPROVED** [review]

**Comment by jolheiser** [comment]
Regarding lunny's concerns, I am working on a PR that should make the dashboard more helpful contextually when it comes to PR reviews.  I'll take this PR's change into account.
dashboard-review

File: models/review.go [hunk 1]
```diff
@@ -272,7 +272,7 @@ func SubmitReview(doer *User, issue *Issue, reviewType ReviewType, content strin
 		if err := review.loadCodeComments(sess); err != nil {
 			return nil, nil, err
 		}
-		if len(review.CodeComments) == 0 && len(strings.TrimSpace(content)) == 0 {
+		if reviewType != ReviewTypeApprove && len(review.CodeComments) == 0 && len(strings.TrimSpace(content)) == 0 {
```
Comment (guillep2k) [inline models/review.go:275]:
```suggestion
		if (reviewType != ReviewTypeApprove || len(review.CodeComments) == 0) && len(strings.TrimSpace(content)) == 0 {
```
~~Maybe?~~
EDIT: Never mind

**Comment by zeripath** [comment]
make lg-tm work

**Comment by jolheiser** [comment]
FYI this doesn't affect `1.10.0` so a backport isn't needed.

**Pull request merged**